connect_four.py

```diff
@@ -133,7 +136,7 @@ def new_turn(self, player: str, color: str) -> None:
         new_row, new_col = self.board.num_rows - len(self.board.columns[col_i]), col_i 
```

Comment:
Same thing applies here, avoid mutation of the columns varaible

Reply:
This can go in the insert new  piece method too

Reply:
Whoops, my b. I'll change that rn.

Reply:
This line below can go in the insert_new_piece method as well. 
new_row, new_col = self.board.num_rows - len(self.board.columns[col_i]), col_i 

And I think the API could take just the column instead of the new row new col.

One reason you would want to take the column as input instead of the row, col combination is that you want to create APIs that could be misused as little as possible. In this case, if you provide the row, col combination, it allows the game loop access to mutate any area of the board. If on the other hand, you just provide the col as an API, you don't have as much power to call the wrong apis.

Also, another way of thinking about it: when you play the actual game of connect four, you don't have the ability to place a tile in the grid at a given location. You put the tile inside the column and the mechanical mechanism takes care of ensuring that the tile goes to the right row, col combination. So the idea is that the logic for determining the row is abstracted away from the user/game loop. 

Let me know if there's any confusion around this.